src/RedeemFei.sol

```diff
@@ -0,0 +1,160 @@
+    function approve(address spender, uint256 amount) external returns (bool);
+    function transfer(address to, uint256 amount) external;
+}
+
+interface ILendingPool {
+    /**
+     * @dev Withdraws an `amount` of underlying asset from the reserve, burning the equivalent aTokens owned
+     * E.g. User has 100 aUSDC, calls withdraw() and receives 100 USDC, burning the 100 aUSDC
+     * @param asset The address of the underlying asset to withdraw
+     * @param amount The underlying amount to be withdrawn
+     *   - Send the value type(uint256).max in order to withdraw the whole aToken balance
+     * @param to Address that will receive the underlying, same as msg.sender if the user
+     *   wants to receive it on his own wallet, or a different address if the beneficiary is a
+     *   different wallet
+     * @return The final amount withdrawn
+     **/
+    function withdraw(
+        address asset,
+        uint256 amount,
+        address to
+    ) external returns (uint256);
+
+    /**
+     * @dev Deposits an `amount` of underlying asset into the reserve, receiving in return overlying aTokens.
+     * - E.g. User deposits 100 USDC and gets in return 100 aUSDC
+     * @param asset The address of the underlying asset to deposit
+     * @param amount The amount to be deposited
+     * @param onBehalfOf The address that will receive the aTokens, same as msg.sender if the user
+     *   wants to receive them on his own wallet, or a different address if the beneficiary of aTokens
+     *   is a different wallet
+     * @param referralCode Code used to register the integrator originating the operation, for potential rewards.
+     *   0 if the action is executed directly by the user, without any middle-man
+     **/
+    function deposit(
+        address asset,
+        uint256 amount,
+        address onBehalfOf,
+        uint16 referralCode
+    ) external;
+}
+
+interface IEcosystemReserveController {
+    /**
+     * @notice Proxy function for ERC20's approve(), pointing to a specific collector contract
+     * @param collector The collector contract with funds (Aave ecosystem reserve)
+     * @param token The asset address
+     * @param recipient Allowance's recipient
+     * @param amount Allowance to approve
+     **/
+    function approve(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+
+    /**
+     * @notice Proxy function for ERC20's transfer(), pointing to a specific collector contract
+     * @param collector The collector contract with funds (Aave ecosystem reserve)
+     * @param token The asset address
+     * @param recipient Transfer's recipient
+     * @param amount Amount to transfer
+     **/
+    function transfer(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+}
+
+event PsmRedeem();
+event notPsmRedeem();
+
+contract RedeemFei is IProposalGenericExecutor {
```

Comment:
Can you link the the governance post and snapshot links under the @dev natspec? Also add @notice natspec describing what this payload is doing and add @author natspec as Llama.